subgraphs/compound-forks/protocols/dforce/src/mapping.ts

```diff
@@ -152,7 +151,7 @@ export function handleMarketAdded(event: MarketAdded): void {
       underlyingName = "Maker";
       underlyingSymbol = "MKR";
       underlyingDecimals = 18;
-    } else if (underlyingTokenAddr.toHexString() == DF_ADDRESS) {
+    } else if (underlyingTokenAddr == DF_ADDRESS) {
```

Comment:
Hmm, are you sure this works? Usually on JS/TS, comparing objects or arrays only returns true when you are comparing to references to the same instance:
```
const a = {};
const b = {}
let c = a;
a == b // false
a == c // true
```

```

Reply:
I have been doing this myself - I think it works for string?

```
> const a = "0x01234567890abcdef"
undefined
> const b = "0x01234567890abcdef"
undefined
> a == b
true
```

Reply:
Yeah, strings, booleans and numbers are fine. Usually the problem is with variables whose value is a reference: objects, classes, arrays…

Taking a look at the `Address` implementation, it extends from `ByteArray`, which overloads the `==` operator: https://github.com/graphprotocol/graph-ts/blob/main/common/collections.ts#L259

So whenever you do `Address == Address`, what happens under the hood is that it runs `Address.equals(address)`. It should work then 🤔 

I’d suggest not relying on this in general, since you cannot expect it to always be the case. Safer to call `.equals` explicitly, you always know what to expect from that.

Reply:
Good point! I learned something. thx!

Reply:
Good catch! fixed